ra2ce/graph/hazard.py

```diff
@@ -516,6 +517,42 @@ def join_table(
         )  # Check if this is the right name
         return graph
 
+    def get_point_hazard_from_network(
+        self, points: gpd.GeoDataFrame, network: gpd.GeoDataFrame
+    ) -> gpd.GeoDataFrame:
+        """get hazard informations for points at nearest network locations.
+
+        Args:
+            points (gpd.GeoDataFrame): points of interest, will be snapped to the network by nearest distance method.
+            network (gpd.GeoDataFrame): network of interest
+
+        Returns:
+            gpd.GeoDataFrame: points with relation to network and hazard informations
+
+        Raise:
+            error: if node id columns do not exisit in the network
+        """
+
+        if "node_A" in network.columns:  # shapefiles
+            network["edge_fid"] = [
+                f"{na}_{nb}" for na, nb in network[["node_A", "node_B"]].values
+            ]
+        elif "u" in network.columns:  # osm
+            network["edge_fid"] = [
+                f"{na}_{nb}" for na, nb in network[["u", "v"]].values
```

Comment:
These are implicity assuming that if "node_A" or "u" are in the columns indices then their "paired" keys will as well be there. Why not doing directly said checks? You ensure then full compatibility.
```python
if all(c_idx in network.columns for c_idx in ["node_A", "node_B"]):
```